dbt_sugar/core/task/doc.py

```diff
@@ -48,28 +47,20 @@ def run(self) -> int:
 
         dbt_credentials = self.load_dbt_credentials()
         type_of_connection = dbt_credentials.get("type", "")
+        self.connector = BaseConnector(
```

Comment:
I'm not sure I like that very much it breaks the ability for people to easily write plugins for other databases as they would have to keep on adding if statements to the `BaseConnector` and I would think it is better to write a factory for this as I really want us to quickly allow for people to write their own connectors as plugins.

What made you decide on this change of architecture?